spec-main/extensions-spec.ts

```diff
@@ -297,30 +298,30 @@ describe('chrome extensions', () => {
     });
 
     it('does not take precedence over Electron webRequest - http', async () => {
-      return new Promise<void>((resolve) => {
-        (async () => {
-          customSession.webRequest.onBeforeRequest((details, callback) => {
-            resolve();
-            callback({ cancel: true });
-          });
-          await w.loadURL(url);
+      const run = async (resolve: Function) => {
+        customSession.webRequest.onBeforeRequest((details, callback) => {
+          resolve();
+          callback({ cancel: true });
+        });
 
-          await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
-          fetch(w.webContents, url);
-        })();
-      });
+        await w.loadURL(url);
+        await customSession.loadExtension(path.join(fixtures, 'extensions', 'chrome-webRequest'));
+        fetch(w.webContents, url);
+      };
+
+      await doAsyncWork(run);
```

Comment:
... why is this wrapped in a Promise in the first place? can we just remove all these wrappers and write the code straightforwardly?

Reply:
(see above comment)